### ISSUE TITLE
Correct python requirement in twisted/topfiles/setup.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5060]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5060|
|Type|defect|
|Created|2011-04-20 22:24:59Z|
|Branch|https://github.com/twisted/twisted/tree/minimum-supported-2.5-5060|

[twisted/topfiles/setup.py](twisted/topfiles/setup.py) forces Python 2.3 to be the minimum but this should be 2.4 like the check in [twisted/__init__.py](twisted/__init__.py). Perhaps it should be centralized somewhere, available to other packages as well, like the copyright/version string?

<details><summary>Searchable metadata</summary>

```
trac-id__5060 5060
type__defect defect
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_minimum_supported_2_5_5060 branches/minimum-supported-2.5-5060
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1303338299000000 1303338299000000
changetime__1321802398000000 1321802398000000
version__None None
owner__exarkun exarkun
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

Another python2.3 reference can be found in [twisted/python/dist.py#L80](twisted/python/dist.py#L80).

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [32951]) Branching to 'minimum-supported-2.5-5060'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [32979]) Drop some Python 2.2 support code, bump the documentation and install-ish time checks to Python 2.5

refs [#5060](https://github.com/twisted/twisted/issues/5060)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

Made some changes in the branch.  Didn't try to refactor the duplication.  That'd be nice, but changing this is a very infrequent event so I don't quite feel motivated enough to figure out how to do that.

Tentatively added a .removal news fragment; maybe that's how we want to express this change to users, maybe not.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess commented|
|-|-|

Ticket [4962](http://twistedmatrix.com/trac/ticket/4962) describes when to drop support for Python 2.4, which currently says announcing it in 11.1 and dropping in the subsequent release. Reviewing in anticipation of that:

 * Your changes look good.
 * There are a lot of old references to Python 2.2., 2.3, and 2.4 that should get looked at in a couple of new tickets. I opened [5385](http://twistedmatrix.com/trac/ticket/5385), [5386](http://twistedmatrix.com/trac/ticket/5386), and [5387](http://twistedmatrix.com/trac/ticket/5387) to track these. There are also some comments about Linux 2.2 that could probably also go away.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [33200]) Merge minimum-supported-2.5-5060

Author: exarkun
Reviewer: jesstess
Fixes: [#5060](https://github.com/twisted/twisted/issues/5060)

Change overall version checks and install documentation to require Python 2.5,
reflecting the decision to make that the minimum required version of Python
from the next release onward.
